lambda/index.js

```diff
@@ -111,45 +116,145 @@ function putRecordToDynamo ({jobs, record, dynamo, isoDate, isProd, callback, lo
     const errorCount = attributes.errorCount
         ? parseInt(item.Attributes.errorCount.N, 10) : 0;
+    const error = errorParser.parse(attributes.messageText ? attributes.messageText.S : 'unknown error', MAX_ERROR_LENGTH);
     const frontId = attributes.frontId ? attributes.frontId.S : 'unknown';
-    const error = attributes.messageText ? attributes.messageText.S : 'unknown error';
+
     const isLive = attributes.stageName ? attributes.stageName.S === 'live' : false;
+    if (isLive && errorCount >= ERROR_THRESHOLD) {
+
+        dynamo.getItem({
+            TableName: ERRORS_TABLE_NAME,
+            Key: {
+                error: {
+                    S: error
+                }
+            }
+        }, (err, data) => {
+
+            if (err) {
+                logger.error('Error while fetching error item with message ', err);
+                callback();
+            }
+
+            if (data && data.Item) {
+
+                const affectedFronts = new Set(data.Item.affectedFronts.SS);
+                affectedFronts.add(frontId);
+
 
-    if (isProd && isLive && errorCount >= ERROR_THRESHOLD) {
-        logger.log('Notifying pagerduty');
-        return post({
-            url: 'https://events.pagerduty.com/generic/2010-04-15/create_event.json',
-            body: JSON.stringify({
-                // eslint-disable-next-line camelcase
-                service_key: config.pagerduty.key,
-                // eslint-disable-next-line camelcase
-                event_type: 'trigger',
-                // eslint-disable-next-line camelcase
-                incident_key: frontId,
-                description: `Front ${frontId} failed pressing`,
-                details: {
-                    front: frontId,
-                    stage: attributes.stageName ? attributes.stageName.S : 'unknown',
-                    count: errorCount,
-                    error: error
-                },
-                client: 'Press monitor',
-                // eslint-disable-next-line camelcase
-                client_url: `${config.facia[STAGE].path}/troubleshoot/stale/${frontId}`
-            })
+                const updateErrorData = {
+                    TableName: ERRORS_TABLE_NAME,
+                    Key: {
+                        error: {
+                            S: error
+                        }
+                    },
+                    AttributeUpdates: {
+                        lastSeen: {
+                            Value: {
+                                N: today.valueOf().toString()
+                            },
+                            Action: 'PUT'
+                        },
+                        affectedFronts: {
+                            Value: {
+                                SS: Array.from(affectedFronts)
+                            },
+                            Action: 'PUT'
+                        }
+                    }
+                };
```

Comment:
As someone who's not seen that much of the JS / Dynamo API it's a bit difficult to parse with all of this boilerplate. Is it worth refactoring things like building an update object into a function?

Reply:
I did that and found a little bug with the way I was recording the affected fronts while looking at it. There is also an AWS document client for javascript which would make all of this a bit easier but I didn't want to use because I didn't want to start rewriting the whole lambda.

Reply:
Nice 👍 and yeah I agree re: not rewriting the whole lambda!